frontend/src/components/Branch/BranchMergeView/BranchMergeView.vue

```diff
@@ -198,27 +195,42 @@ const handleMergeBranch = async () => {
       etag: "",
       validateOnly: false,
     });
-    if (state.deleteBranchAfterMerged) {
+
+    pushNotification({
+      module: "bytebase",
+      style: "SUCCESS",
+      title: t("branch.merge-rebase.merge-succeeded"),
+    });
+    if (post === "DELETE") {
       await branchStore.deleteBranch(head.name);
       pushNotification({
         module: "bytebase",
         style: "SUCCESS",
         title: t("common.deleted"),
       });
-    }
 
-    pushNotification({
-      module: "bytebase",
-      style: "SUCCESS",
-      title: t("branch.merge-rebase.merge-succeeded"),
-    });
+      emit("merged", mergedBranch, head.name, undefined);
+      return;
+    }
+    if (post === "REBASE") {
```

Comment:
Use delete and create for this? Not sure confident about our rebase, experience-wise.

Reply:
We don't literally "delete" the branch so we can't actually deleted and re-create it since the branch id will conflict

Reply:
For branch, we do physical deletion.

Reply:
I think it's alright for keeping the current way. Let's give it a try.